MFVNeutralino/plugins/Vertexer.cc

```diff
@@ -1,891 +1,1778 @@
+                printf("sv1'idx: %lu with ntrack after: %u",vtxidx1, sv1.nTracks());
+              }
+            }
+          }
+        }
+      }
+    }
+
+    std::set<int> trackicity;
+    for (v[0] = vertices->begin(); v[0] != vertices->end(); ++v[0]) {
+      track_set tracks[2];
+      ivtx[0] = v[0] - vertices->begin();
+      tracks[0] = vertex_track_set(*v[0]);
+
+      if (tracks[0].size() < 2) {
+        if (verbose)
+          printf("resolve_shared_jets: vertex-0 #%lu is down to one track, junking it\n", ivtx[0]);
+        v[0] = vertices->erase(v[0]) - 1;
+        ++n_output_aftersharedjets_onetracks;
+        continue;
+      }
+
+      if (tracks[0].size() < v[0]->nTracks()) {
+        std::cout << "duplicate tracks in vertexer" << std::endl;
+        std::vector<reco::TransientTrack> sv_nonduplicate_ttks;
+        for (const reco::TrackRef& itk : vertex_track_set(*v[0])) {
+          if (itk.isNonnull())
+            sv_nonduplicate_ttks.push_back(tt_builder->build(itk));
+        }
+
+        for (const TransientVertex& tv : kv_reco_dropin(sv_nonduplicate_ttks))
+          *v[0] = reco::Vertex(tv);
+      }
+    }
+  }
+
+  if (resolve_shared_jets && histos_output_aftersharedjets) {
+    fillCommonOutputHists(vertices, fake_bs_vtx, tt_builder, stepEnum::aftersharedjets);
+    h_output_aftersharedjets_n_onetracks->Fill(n_output_aftersharedjets_onetracks);
+  }
+
+  //////////////////////////////////////////////////////////////////////
+  // Put the output.
+  //////////////////////////////////////////////////////////////////////
+
+  finish(event, seed_tracks, std::move(vertices), std::move(vpeffs), vpeffs_tracks);
+}
+
+std::pair<bool, std::vector<std::vector<size_t>>> MFVVertexer::sharedjets(const int vtx0idx, const int vtx1idx, const std::vector < std::vector<int>>& sv_match_tracktojet_which_jetidx, const std::vector < std::vector<int>>& sv_match_tracktojet_which_trkidx) {
+
+  bool shared_jet = std::find_first_of(sv_match_tracktojet_which_jetidx[vtx0idx].begin(), sv_match_tracktojet_which_jetidx[vtx0idx].end(), sv_match_tracktojet_which_jetidx[vtx1idx].begin(), sv_match_tracktojet_which_jetidx[vtx1idx].end()) != sv_match_tracktojet_which_jetidx[vtx0idx].end();
+
+  std::vector<std::vector<size_t>> sv_lonesharedtrack_which_trkidx;
+  std::vector<size_t> sv0_lonesharedtrack_which_trkidx;
+  std::vector<size_t> sv1_lonesharedtrack_which_trkidx;
+
+  if (shared_jet) {
+
+    int nsharedjets = 0;
+
+    std::vector<std::vector<int>> sv_match_tracktojet_which_temp_jetidx = sv_match_tracktojet_which_jetidx;
+    std::vector<std::vector<int> >sv0_match_sharedjettotrack_which_trkidx;
+    std::vector<std::vector<int> >sv1_match_sharedjettotrack_which_trkidx;
+
+    std::vector<int> sv0_match_tracktojet_which_jetidx = sv_match_tracktojet_which_jetidx[vtx0idx];
+    std::vector<int> sv0_match_tracktojet_which_trkidx = sv_match_tracktojet_which_trkidx[vtx0idx];
+    std::vector<int> sv0_match_sharedjettotrack_which_temp_trkidx;

```

Comment:
Could we shorten some of the names here (and others)? Maybe `sv0_match_tracktojet_which_jetidx` could simply be `sv0_match_jetidx`? Similarly would be good for others, to make it a bit more readable.

Reply:
yep, done.